src/ripple/overlay/impl/Message.cpp

```diff
@@ -41,18 +42,53 @@ Message::Message (::google::protobuf::Message const& message, int type)
 
     mBuffer.resize (headerBytes + messageBytes);
 
-    auto ptr = mBuffer.data();
+    auto set_header = [](uint8_t *ptr, decltype(messageBytes) messageBytes, int type, bool compressed=false,
+            uint8_t compr_algorithm=ripple::compression::Algorithm::LZ4) {
+        uint8_t compression = (compressed?0xF0:0) & (0x80 | (compr_algorithm << 4));
+        *ptr++ = static_cast<std::uint8_t>(((messageBytes >> 24) | compression) & 0xFF);
+        *ptr++ = static_cast<std::uint8_t>((messageBytes >> 16) & 0xFF);
+        *ptr++ = static_cast<std::uint8_t>((messageBytes >> 8) & 0xFF);
+        *ptr++ = static_cast<std::uint8_t>(messageBytes & 0xFF);
 
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 24) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 16) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 8) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>(messageBytes & 0xFF);
+        *ptr++ = static_cast<std::uint8_t>((type >> 8) & 0xFF);
+        *ptr = static_cast<std::uint8_t> (type & 0xFF);
+    };
 
-    *ptr++ = static_cast<std::uint8_t>((type >> 8) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t> (type & 0xFF);
+    set_header(mBuffer.data(), messageBytes, type);
 
     if (messageBytes != 0)
-        message.SerializeToArray(ptr, messageBytes);
+        message.SerializeToArray(mBuffer.data() + headerBytes, messageBytes);
+
+    bool compressible = compression_enabled &&
+            (type == protocol::mtMANIFESTS || type == protocol::mtENDPOINTS ||
+             type == protocol::mtTRANSACTION || type == protocol::mtGET_LEDGER || type == protocol::mtLEDGER_DATA ||
+             type == protocol::mtGET_OBJECTS || type == protocol::mtVALIDATORLIST) &&
+            messageBytes > 70;
```

Comment:
What do you think about a lambda and a switch to reduce all the `type ==` noise? Something like:
```

    bool const compressible = [&]{
        if (!compression_enabled || messageBytes <= 70)
            return false;
        switch(type)
        {
            case protocol::mtMANIFESTS:
            case protocol::mtENDPOINTS:
            case protocol::mtTRANSACTION:
            case protocol::mtGET_LEDGER:
            case protocol::mtLEDGER_DATA:
            case protocol::mtGET_OBJECTS:
            case protocol::mtVALIDATORLIST:
                return true;
            default:
                return false;
        }
    }();
```

Reply:
Done.